features/reset/reset_and_delete/feature.go

```diff
@@ -0,0 +1,174 @@
+	foundID, err := findSecondWFTComplete(iter)
+	if err != nil {
+		return run, err
+	}
+
+	// Reset it
+	r.Log.Info("Resetting workflow", "ToEventId", foundID)
+	_, err = r.Client.ResetWorkflowExecution(ctx, &workflowservice.ResetWorkflowExecutionRequest{
+		Namespace: r.Namespace,
+		WorkflowExecution: &common.WorkflowExecution{
+			WorkflowId: run.GetID(),
+			RunId:      run.GetRunID(),
+		},
+		Reason:                    "because I feel like it",
+		WorkflowTaskFinishEventId: foundID,
+	})
+	if err != nil {
+		return nil, err
+	}
+
+	// Drive then finish the now-reset workflow
+	newRun := r.Client.GetWorkflow(ctx, run.GetID(), "")
+	err = r.Client.SignalWorkflow(ctx, newRun.GetID(), "", SignalName, "")
+	if err != nil {
+		return nil, err
+	}
+	err = r.Client.SignalWorkflow(ctx, newRun.GetID(), "", SignalName, "finish")
+	if err != nil {
+		return nil, err
+	}
+
+	return newRun, nil
+}
+
+func CheckResult(ctx context.Context, r *harness.Runner, run client.WorkflowRun) error {
+	// The reset workflow should have finished normally
+	err := r.CheckResultDefault(ctx, run)
+	if err != nil {
+		return err
+	}
+
+	// Now delete it. This can fail initially with "workflow state is not ready"
+	err = r.DoUntilEventually(ctx, 1*time.Second, 1*time.Minute, func() bool {
+		_, err = r.Client.WorkflowService().DeleteWorkflowExecution(ctx, &workflowservice.DeleteWorkflowExecutionRequest{
+			Namespace:         r.Namespace,
+			WorkflowExecution: &common.WorkflowExecution{WorkflowId: run.GetID(), RunId: run.GetRunID()},
+		})
+		return err == nil
+	})
+	// Make sure it's actually gone, since deletion is async
+	err = r.DoUntilEventually(ctx, 1*time.Second, 3*time.Minute, func() bool {
+		_, err := r.Client.DescribeWorkflowExecution(ctx, run.GetID(), run.GetRunID())
+		var notFoundErr *serviceerror.NotFound
+		if errors.As(err, &notFoundErr) {
+			return true
+		}
+		return false
+	})
+	if err != nil {
+		return err
+	}
+
+	// Verify original run can still be interacted with
+	origRun := r.Client.GetWorkflow(ctx, run.GetID(), originalRunID)
+	r.Log.Info("Checking history of original run", "RunID", origRun.GetRunID())
+	history := r.Client.GetWorkflowHistory(ctx, origRun.GetID(), origRun.GetRunID(), false, enums.HISTORY_EVENT_FILTER_TYPE_ALL_EVENT)
+	var next *historyProto.HistoryEvent
+	for history.HasNext() {
+		next, err = history.Next()
+		if err != nil {
+			break
+		}
+	}
+	if next.GetEventType() != enums.EVENT_TYPE_WORKFLOW_EXECUTION_TERMINATED {
+		return errors.New("expected last event to be terminated")
+	}
+	// Ensure original run is findable via visibility & has correct status
+	resp, err := r.Client.ListWorkflow(ctx, &workflowservice.ListWorkflowExecutionsRequest{
+		Namespace: r.Namespace,
+		Query:     "WorkflowId = '" + origRun.GetID() + "'",
+	})
+	if err != nil {
+		return err
+	}
+	r.Assert.Equal(1, len(resp.GetExecutions()))
```

Comment:
I personally like `r.Require` instead of `r.Assert` because it fails immediately, but difference is often negligible